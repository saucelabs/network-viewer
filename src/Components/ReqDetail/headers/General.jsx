import React from 'react';
import PropTypes from 'prop-types';

import Styles from './../Headers.styles.scss';
import { GENERAL_HEADERS } from '../../../constants';

const General = ({ data }) => (
  <div className={Styles['header-detail']}>
    {Object.entries(GENERAL_HEADERS).map(([dataKey, { key, name }]) => (
      <p
        key={dataKey}
        className={Styles['info-row']}
      >
        <span className={Styles['info-caption']}>
          {`${name}:`}
        </span>
        <span className={Styles['info-value']}>
          {key === 'status' && data.error ? data.error : data[key]}
        </span>
      </p>
    ))}
  </div>
);

General.propTypes = {
  data: PropTypes.object,
};

General.defaultProps = {
  data: null,
};

export default General;
