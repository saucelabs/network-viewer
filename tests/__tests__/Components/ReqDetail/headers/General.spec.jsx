import React from 'react';
import { shallow } from 'enzyme';

import General from './../../../../../src/Components/ReqDetail/headers/General';

describe('General', () => {
  const props = {
    data: {},
  };

  it('renders without crashing', () => {
    const element = shallow(<General {...props} />);
    expect(element).toMatchSnapshot();
  });

  it('renders intercept error', () => {
    const element = shallow(<General data={{ error: 'ERR_TIMED_OUT' }} />);
    expect(element).toMatchSnapshot();
  });
});
