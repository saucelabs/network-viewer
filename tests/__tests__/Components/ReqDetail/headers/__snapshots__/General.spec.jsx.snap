// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`General renders intercept error 1`] = `
<div>
  <p
    key="url"
  >
    <span>
      Request URL:
    </span>
    <span />
  </p>
  <p
    key="method"
  >
    <span>
      Request Method:
    </span>
    <span />
  </p>
  <p
    key="status"
  >
    <span>
      Status Code:
    </span>
    <span>
      ERR_TIMED_OUT
    </span>
  </p>
  <p
    key="serverIPAddress"
  >
    <span>
      Remote Address:
    </span>
    <span />
  </p>
</div>
`;

exports[`General renders without crashing 1`] = `
<div>
  <p
    key="url"
  >
    <span>
      Request URL:
    </span>
    <span />
  </p>
  <p
    key="method"
  >
    <span>
      Request Method:
    </span>
    <span />
  </p>
  <p
    key="status"
  >
    <span>
      Status Code:
    </span>
    <span />
  </p>
  <p
    key="serverIPAddress"
  >
    <span>
      Remote Address:
    </span>
    <span />
  </p>
</div>
`;
